### DISCUSSION TITLE
How is this different from HTMX?

### DISCUSSION THREAD

**Discussion by port19x** [alpinejs/alpine#3732] [4. General]
... is a question many first time visitors will ask themselves now that htmx is exploding in popularity.
Therefore I'm asking this question here, so that an answer can be found by people afraid to ask for themselves.

Is there a synergy going on?
Is there significant overlap?
Do the projects solve entierly different problems and only coincidentally in a similar way?

**Comment by ekwoka** [comment]
It's more that they solve similar related problems in different ways that are better suited to a specific type of interaction.

HTMX is purely about server rendering everything. It makes it easier to do requests that return html and update the DOM on the page.

Alpine is client side reactivity. Meaning the client is handling it's own render changes.

They can work together (and work well together, I have large clients using both together) where Alpine handles some kinds of reactivity that is impractical to pass off to the server (opening a popup for example, incrementing counters, or even working with rest apis over HTML apis), while htmx handles boosting page swaps and form submissions.

They do also work together. There is an official HTMX plugin that allows you to use Alpine.morph as the `swap` style, instead of entirely swapping the full html trashing and creating new DOM nodes, Alpine handles diffing and modifying the live DOM to match the newly provided DOM while not losing Alpine states.

Broadly, they are just different things.
